### PR TITLE
Firmware support for RTR_520 with ESP32

### PR DESCRIPTION
**Pull request by thias15**

### CODE REVIEW

File: firmware/openbot_nano/openbot_nano.ino [hunk 1]
```diff
@@ -235,6 +252,79 @@ const int PIN_PWM_R1 = 9;
 const int PIN_PWM_R2 = 10;
 const int PIN_LED_LI = 4;
 const int PIN_LED_RI = 7;
+//-------------------------RTR_520----------------------//
+#elif (OPENBOT == RTR_520)
```
Comment (quentin-leboutet) [inline firmware/openbot_nano/openbot_nano.ino:256]: I need to redefine the min and max functions to get it to build: 
```
#define max(a, b)                   \
    (                               \
        {                           \
            __typeof__(a) _a = (a); \
            __typeof__(b) _b = (b); \
            _a > _b ? _a : _b;      \
        })

#define min(a, b)                   \
    (                               \
        {                           \
            __typeof__(a) _a = (a); \
            __typeof__(b) _b = (b); \
            _a < _b ? _a : _b;      \
        })
```

**Review by quentin-leboutet — CHANGES_REQUESTED** [review]
The min and max functions need to be redefined so get it to build properly. Once this is done, I confirm that I can control the RTR_520 vehicle.

File: firmware/openbot_nano/openbot_nano.ino [hunk 2]
```diff
@@ -251,10 +341,12 @@ int ctrl_min = (int) 255.0 * VOLTAGE_MIN / VOLTAGE_MAX;
...
 
 #if HAS_SONAR
+#if (OPENBOT != RTR_520)
 #include "PinChangeInterrupt.h"
+#endif
 //Sonar sensor
 const float US_TO_CM = 0.01715;              //cm/uS -> (343 * 100 / 1000000) / 2;
-const unsigned int MAX_SONAR_DISTANCE = -1;  //cm
+const unsigned int MAX_SONAR_DISTANCE = 300;  //cm
 const unsigned int MAX_SONAR_TIME = MAX_SONAR_DISTANCE * 2 * 10 / 343 + 1;
```
Comment (quentin-leboutet) [inline firmware/openbot_nano/openbot_nano.ino:350]: As discussed with Matthias, an alternative solution consists in changing this line to:
```
const unsigned long MAX_SONAR_TIME = (long)(MAX_SONAR_DISTANCE * 2 * 10 / 343 + 1);
```

Comment (thias15) [reply firmware/openbot_nano/openbot_nano.ino]: Updated as discussed.

File: firmware/openbot_nano/openbot_nano.ino (hunk 1 above)
Comment (thias15) [reply firmware/openbot_nano/openbot_nano.ino:256]: This should be fixed now with the fix below.

**Review by quentin-leboutet — APPROVED** [review]

**Pull request merged**